### PR TITLE
widget 2.0: enforce sorting and your folders first always

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers

### CODE REVIEW

File: shared/actions/favorite.js [hunk 1]
```diff
@@ -24,16 +24,24 @@ export function favoriteList (): (dispatch: Dispatch) => void {
...
 
         // Ensure private/public folders exist for us
         if (currentUser && loggedIn) {
           [true, false].forEach(isPrivate => {
-            if (!folders.find(f => f.name === currentUser && f.private === isPrivate)) {
-              folders = [{name: currentUser, private: isPrivate, notificationsOn: false, created: false}, ...folders]
+            const idx = folders.findIndex(f => f.name === currentUser && f.private === isPrivate)
+            let toAdd = {name: currentUser, private: isPrivate, notificationsOn: false, created: false}
+
+            if (idx !== -1) {
```
Comment (chrisnojima) [inline shared/actions/favorite.js]: if we're already in the list let's re-add and remove from the original list

**Comment by chromakode** [comment]
:+1:

File: shared/actions/favorite.js (hunk 1 above)
Comment (MarcoPolo) [inline shared/actions/favorite.js]: you can replace this idx code with a filter and using a similar fn that you pass into findIndex.

**Comment by MarcoPolo** [comment]
👍 % filter suggestion, I think it'll be a little cleaner

**Comment by chrisnojima** [comment]
But I need to know if it is filtered

**Comment by MarcoPolo** [comment]
I mean doing:

```js
folders = [toAdd, ...folders.filter(f => !(f.name === currentUser && f.private === isPrivate))]
```

That's the same thing, no?

**Comment by chrisnojima** [comment]
Kinda except them I'm looking through the array twice

**Comment by chrisnojima** [comment]
after our slack convo using unshift instead

**Pull request merged**